pallets/account-linker/src/util_eth.rs

```diff
@@ -7,15 +11,51 @@ pub fn addr_from_sig(msg: [u8; 32], sig: [u8; 65]) -> Result<[u8; 20], sp_io::Ec
 	Ok(addr)
 }
 
+/// Returns a eth_sign-compatible hash of data to sign.
+/// The data is prefixed with special message to prevent
+/// malicious DApps from using the function to sign forged transactions.
+pub fn eth_data_hash(mut data: Vec<u8>) -> Result<[u8; 32], &'static str> {
+	const MSG_LEN: usize = 51;
+	if data.len() != MSG_LEN {
+		debug::error!("Ethereum message has an unexpected length {} !!! Expected is {}.", data.len(), MSG_LEN);
+		return Err("Unexpected ethereum message length!");
+	}
```

Comment:
These are no longer needed

Reply:
That's something I also would like to discuss, about how and where should we do this length check. I know it's not optimal to put it inside a function in util, rather we need to have it checked in account-linker so that we can update the const MSG_LEN there once our message format changes again.

I have a proposal, that we can have an additional parameter for this function as expect_length and it can vary from different callers for different purposes (in case we reuse it). How do you think?

Reply:
I checked this when I wrote the comment. I think the current code is OK since we have generate_msg func in test which uses hardcoded 51, so if we change the types (so the length changes). The test will no longer pass. 
Of course better to have some comments say around generate_msg to mention what's wrong if these tests fail

Reply:
For your proposal, I think our goal is to make any thing related to runtime as small as possible (we may have a lot in the tests)